src/main/java/org/opensearch/security/filter/OpenSearchRequest.java

```diff
@@ -0,0 +1,78 @@
+package org.opensearch.security.filter;
+
+import java.net.InetSocketAddress;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.net.ssl.SSLEngine;
+
+import org.opensearch.http.netty4.Netty4HttpChannel;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+
+import io.netty.handler.ssl.SslHandler;
+
+/**
+ * Wraps the functionality of RestRequest for use in the security plugin
+ */
+public class OpenSearchRequest implements SecurityRequest {
+
+    protected final RestRequest underlyingRequest;
+
+    OpenSearchRequest(final RestRequest request) {
+        underlyingRequest = request;
+    }
+
+    @Override
+    public Map<String, List<String>> getHeaders() {
+        return underlyingRequest.getHeaders();
+    }
+
+    @Override
+    public SSLEngine getSSLEngine() {
+        if (underlyingRequest == null
+            || underlyingRequest.getHttpChannel() == null
+            || !(underlyingRequest.getHttpChannel() instanceof Netty4HttpChannel)) {
+            return null;
+        }
+
+        final Netty4HttpChannel httpChannel = (Netty4HttpChannel) underlyingRequest.getHttpChannel();
+        SslHandler sslhandler = (SslHandler) httpChannel.getNettyChannel().pipeline().get("ssl_http");
+        if (sslhandler == null && httpChannel.inboundPipeline() != null) {
+            sslhandler = (SslHandler) httpChannel.inboundPipeline().get("ssl_http");
+        }
```

Comment:
What do you think about adding a brief comment here that explains these 4 lines of code:
```java
// We look for Ssl_handler called `ssl_http` in the outbound pipeline of Netty channel first, and if its not
// present we look for it in inbound channel. If its present in neither we return null, else we return the sslHandler.
```